### DISCUSSION TITLE
"definition" renderers tag removed in React-Markdown 6

### DISCUSSION THREAD

**Discussion by DaniG2106** [remarkjs/remark#696] [Q&A] [answered]
We are using react-markdown to render some custom charts for our project. In v5 we used the "definition" tag to render the charts. As we are now trying to update react-markdown to the newest version, we recognized a lot of changes and unfortunately the removal of that tag. 
At the moment we are looking for an easier solution than reworking the complete code. One idea from our side was to use the html tag in future. Does anyone know why "definition" is removed and if there is an easier solution? Most of the other tags just received new names.
There is a hint in the changelog mentioning:
> Resource ([text](url)) and reference ([text][id]) style links and images (and their definitions) are now resolved and treated the same

Maybe someone also knows how to interpret that.

Thank you for your help!

**Comment by ChristianMurphy** [comment, marked as answer]
the new flow is

```
                                                           react-markdown
+---------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                                                             |
|            +----------+        +----------------+        +---------------+       +----------------+       +--------------+                  |
|            |          |        |                |        |               |       |                |       |              |                  |
| -markdown->+  remark  +-mdast->+ remark plugins +-mdast->+ remark-rehype +-hast->+ rehype plugins +-hast->+  components  +-react elements-> |
|            |          |        |                |        |               |       |                |       |              |                  |
|            +----------+        +----------------+        +---------------+       +----------------+       +--------------+                  |
|                                                                                                                                             |
+---------------------------------------------------------------------------------------------------------------------------------------------+
```

the custom mdast nodes (like directives) need a plugin to convert it to hast.
There's an example of how to do this in the readme of `remark-directive` (https://github.com/remarkjs/remark-directive#use) with the `htmlDirectives` plugin, which can be used as an example of a `remarkPlugin` for `react-markdown`
